File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hashmap;
        vector<int> solution;
        for(int i=0;i<nums.size();i=i+1)
        {
            if(hashmap.count(nums[i]))
            {
                solution.push_back(i);
                solution.push_back(hashmap[nums[i]]);
            }
            else
            {
                hashmap.insert(make_pair(target-nums[i],i));
            }
        }
        return solution;
    }
};

File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int[] result=new int[2];

        HashMap<Integer,Integer> map = new HashMap<Integer, Integer>();
        
        for(int i=0;i<nums.length;i=i+1)
        {
            if(map.get(nums[i]) == null)
            {
                map.put(target-nums[i],i);
            }else
            {
                result[0] = map.get(nums[i]);
                result[1] = i;
            }
                
        }
        return result;
    }
}